packages/itwinui-react/src/core/Menu/MenuItem.tsx

```diff
@@ -111,6 +112,7 @@ export const MenuItem = React.forwardRef((props, forwardedRef) => {
   }
 
   const parentMenu = React.useContext(MenuContext);
+  const tileMoreOptionsContext = React.useContext(TileMoreOptionsContext);
```

Comment:
`Tile` depends on `Menu`/`MenuItem` but the vice-versa is not true, so it would be nice to decouple this component from `Tile`.

We could create a context in `Menu`/`MenuItem` such that when it's set (to `true`), the `Menu` will automatically close. Then we can provide this context from `Tile.MoreOptions`.

As a bonus, this approach would allow `Tile.MoreOptions` to stay uncontrolled (i.e. it no longer needs to pass `visible` to `DropdownMenu`).

Reply:
Instead of manually inheriting the context value across all `Menu`s, it might be simpler to provide a single context value for the whole tree in `DropdownMenu`.

In the next minor release, we could even offer this as a prop. I can imagine it being useful for when `menuItems` is used with a React node (instead of a `(close) => []` function).

Reply:
1. Remove `topmostClose` and `MenuItemCloseContext` from `Menu.tsx`.
2. Create a simple boolean context in `DropdownMenu.tsx`.
3. Use the context directly in `MenuItem.tsx`.

If you think about it, "closing" is a feature of the `DropdownMenu`, not the `Menu`, so it makes sense to move the context up there.